tuning/utils/preprocessing_utils.py

```diff
@@ -355,6 +354,14 @@ def load_hf_dataset_from_jsonl_file(
     if input_field_name == output_field_name:
         raise ValueError("Input field name and output field name should not match!")
 
+    file_extension = os.path.splitext(data_path)[-1].lower()
+    if file_extension == ".jsonl":
+        file_format = "jsonl"
+    elif file_extension == ".json":
+        file_format = "json"
+    else:
+        raise ValueError("Unsupported file format! Use 'json' or 'jsonl'.")
```

Comment:
This can be refactored to be at the bottom and do the if statements together:
```py
    file_extension = os.path.splitext(data_path)[-1].lower()
    if file_extension == ".jsonl":
        return Dataset.from_generator(get_jsonl_object)
    elif file_extension == ".json":
        return Dataset.from_generator(get_json_object)
    else:
        raise ValueError("Unsupported file format! Use 'json' or 'jsonl'.")
```

Reply:
Absolutely - very nice!